src/arkode/arkode_mristep.c

```diff
@@ -1021,10 +1019,7 @@ int mriStep_Init(void* arkode_mem, int init_type)
   {
     /* enforce use of arkEwtSmallReal if using a fixed step size for
        an explicit method and an internal error weight function */
-    reset_efun = SUNTRUE;
-    if (step_mem->implicit_rhs) { reset_efun = SUNFALSE; }
-    if (ark_mem->user_efun) { reset_efun = SUNFALSE; }
-    if (reset_efun)
+    if (!step_mem->implicit_rhs && !ark_mem->user_efun)
```

Comment:
Please revert this change, since this logic will be even more complicated when MRI adaptivity is enabled.